teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorClientOptions.java

```diff
@@ -49,16 +49,29 @@ public class ValidatorClientOptions {
       ValidatorConfig.DEFAULT_FAILOVERS_SEND_SUBNET_SUBSCRIPTIONS_ENABLED;
 
   @CommandLine.Option(
-      names = {"--Xbeacon-node-event-stream-syncing-status-query-period"},
+      names = {"--Xbeacon-nodes-syncing-status-query-period"},
       paramLabel = "<INTEGER>",
       description =
-          "How often (in milliseconds) will the syncing status of the Beacon Node used for event streaming be queried. If the node is not synced or the query fails, the Validator Client will try to connect to an event stream of one of the failover Beacon Nodes if such are configured. "
-              + "If the primary Beacon Node has synced successfully after a failover, a reconnection to the primary Beacon Node event stream will be attempted.",
+          "How often (in milliseconds) will the syncing status of the configured Beacon Nodes be queried. "
+              + "The result of the query would be used to determine if a Validator Client should send requests to a Beacon Node or failover immediately. "
+              + "This option is only applicable if there are multiple Beacon Nodes configured.",
       hidden = true,
       showDefaultValue = CommandLine.Help.Visibility.ALWAYS,
       arity = "1")
-  private long beaconNodeEventStreamSyncingStatusQueryPeriod =
-      ValidatorConfig.DEFAULT_BEACON_NODE_EVENT_STREAM_SYNCING_STATUS_QUERY_PERIOD.toMillis();
+  private long beaconNodesSyncingStatusQueryPeriod =
+      ValidatorConfig.DEFAULT_BEACON_NODES_SYNCING_STATUS_QUERY_PERIOD.toMillis();
+
+  @CommandLine.Option(
+      names = {"--Xbeacon-node-event-stream-readiness-check-period"},
+      paramLabel = "<INTEGER>",
+      description =
+          "How often (in milliseconds) will the readiness of the Beacon Node used for event streaming be checked. If the node is not ready, the Validator Client will try to connect to an event stream of one of the failover Beacon Nodes if such are configured. "
+              + "If the primary Beacon Node becomes ready again after a failover, a reconnection to the primary Beacon Node event stream will be attempted.",
+      hidden = true,
+      showDefaultValue = CommandLine.Help.Visibility.ALWAYS,
+      arity = "1")
+  private long beaconNodeEventStreamReadinessCheckPeriod =
+      ValidatorConfig.DEFAULT_BEACON_NODE_EVENT_STREAM_READINESS_CHECK_PERIOD.toMillis();
```

Comment:
im not sure that we need separate frequency for event stream checking, i would have thought we could just fail over to the new BN including event stream... having a separate event stream to the BN you're connected to may lead to a level of chaos..

Reply:
> im not sure that we need separate frequency for event stream checking, i would have thought we could just fail over to the new BN including event stream... having a separate event stream to the BN you're connected to may lead to a level of chaos..

Ideally yes, but problem is a beacon node may be not synced (EL offline for example) and in that case the event stream won't throw an exception and a failover won't be triggered unless the syncing status is checked frequently